inference-engine/tests/functional/plugin/shared/include/single_layer_tests/eltwise.hpp

```diff
@@ -7,44 +7,53 @@
 #include <gtest/gtest.h>
 
 #include <map>
+#include <functional_test_utils/layer_test_utils.hpp>
 
 #include "common_test_utils/common_layers_params.hpp"
 #include "common_test_utils/common_utils.hpp"
 #include "common_test_utils/test_common.hpp"
 #include "common_test_utils/test_constants.hpp"
 #include "ie_core.hpp"
 
-namespace EltwiseTestNamespace {
-
-    using ParameterInputIdx = int;
-    enum class InputLayerType {
-        CONSTANT,
-        PARAMETER
-    };
-    enum class EltwiseOpType {
-        ADD,
-        SUBSTRACT,
-        MULTIPLY
-    };
-    const char* InputLayerType_to_string(InputLayerType lt);
-    const char* EltwiseOpType_to_string(EltwiseOpType eOp);
-}// namespace EltwiseTestNamespace
+namespace LayerTestsDefinitions {
+namespace EltwiseParams {
+enum class InputLayerType {
+    CONSTANT,
+    PARAMETER
+};
+
+enum class OpType {
+    SCALAR,
+    VECTOR
+};
+
+enum class EltwiseOpType {
+    ADD,
+    SUBSTRACT,
+    MULTIPLY
+};
+
+const std::string InputLayerType_to_string(InputLayerType lt);
+const std::string EltwiseOpType_to_string(EltwiseOpType eOp);
+const std::string OpType_to_string(OpType op);
+} // namespace EltwiseParams
 
 typedef std::tuple<
-    EltwiseTestNamespace::EltwiseOpType,       // eltwise op type
-    EltwiseTestNamespace::ParameterInputIdx,   // primary input idx
-    EltwiseTestNamespace::InputLayerType,      // secondary input type
-    InferenceEngine::Precision,                // Net precision
-    InferenceEngine::SizeVector,               // Input shapes
-    std::string,                               // Device name
-    std::map<std::string, std::string>         // Additional network configuration
-> eltwiseLayerTestParamsSet;
-
-class EltwiseLayerTest : public testing::WithParamInterface<eltwiseLayerTestParamsSet>,
+    std::vector<std::vector<size_t>>,             // input shapes
+    EltwiseParams::EltwiseOpType,                 // eltwise op type
+    EltwiseParams::InputLayerType,                // secondary input type
+    EltwiseParams::OpType,                        // op type
+    InferenceEngine::Precision,                   // Net precision
+    std::string,                                  // Device name
+    std::map<std::string, std::string>            // Additional network configuration
+> EltwiseTestParams;
```

Comment:
it might be logical to put this tumple into EltwiseParams namespace taking the name of the tuple and namespace into consideration

Reply:
I disagree because `EltwiseParams` namespace defines layer params. Test params should be in `LayerTestDefinition` namespace.

Reply:
Ok, this one was just suggestion I wont insist on this one